### PR TITLE
Implement inattentive trait: inability to see if creatures surrounding player sees player

### PR DESCRIPTION
**Pull request by rhmkrutov**
#### Summary
Interface "add trait to hamper player's ability to judge creature's vision and intentions"

#### Purpose of change
I think that players being able to see if monsters sees them or not is quite powerful and after a brief discussion with players on discord awhile ago it was said that this would make a good challenge trait.

#### Describe the solution

If player chooses "Inattentive" trait during character creation, they loose the ability to see if monsters see them or not, as well as ability to judge critter's attitude/intent towards them.

#### Describe alternatives you've considered

n/a

#### Testing

Started a game, teleported to town, seen that monsters see me, debug-mutated the trait, seen that I don't know if monsters see me.

#### Additional context

~~There are two problems I don't really know from which side to tackle:~~
~~1) Tilesets - tilesets which support sees_you flag are a dead giveaway.~~
~~2) Hostility status. It seems that there is no easy way of changing only the UI, so I'd like to hear suggestions on this.~~

There are also ~~two~~ one minor problems I don't really know how to solve:
~~1) Trait name. Unattentive is OKish for non-english-native, but probably somebody would come up with a better name.~~
2) Trait cost. I don't really know if 2 points is a good spot, as on one hand it isn't all that limiting, on the other - many players, me included, rely on this information during raiding quite heavily.

### CODE REVIEW

File: data/json/mutations/mutations.json [hunk 1]
```diff
@@ -207,6 +207,16 @@
     "cancels": [ "BADHEARING" ],
     "hearing_modifier": 1.75
   },
+  { 
+    "type": "mutation",
+    "id": "UNATTENTIVE",
+    "name": { "str": "Unattentive" },
+    "points": -2,
+    "starting_trait": true,
+    "description": "You never paid attention to your surroundings and specifically to people around you. You can't judge intent of critters you see.",
```
Comment (Night-Pryanik) [inline data/json/mutations/mutations.json:216]:
```suggestion
    "description": "You never paid attention to your surroundings and specifically to people around you.  You can't judge intent of critters you see.",
```

File: src/character.cpp [hunk 2]
```diff
@@ -462,6 +462,7 @@ static const trait_id trait_THRESH_PLANT( "THRESH_PLANT" );
 static const trait_id trait_THRESH_SPIDER( "THRESH_SPIDER" );
 static const trait_id trait_TOUGH_FEET( "TOUGH_FEET" );
 static const trait_id trait_TRANSPIRATION( "TRANSPIRATION" );
+static const trait_id trait_UNATTENTIVE( "UNATTENTIVE" );
```
Comment (Night-Pryanik) [inline src/character.cpp:465]: It isn't used anywhere in `character.cpp`, so it should be removed.

File: src/game.cpp [hunk 3]
```diff
@@ -7775,7 +7776,7 @@ game::vmenu_ret game::list_monsters( const std::vector<Creature *> &monster_list
                     Creature *critter = monster_list[iCurMon];
                     const bool selected = iCurMon == iActive;
                     ++iCurMon;
-                    if( critter->sees( u ) ) {
+                    if( critter->sees( u ) && !u.has_trait(trait_UNATTENTIVE) ) {
```
Comment (Night-Pryanik) [inline src/game.cpp:7779]:
```suggestion
                    if( critter->sees( u ) && !u.has_trait( trait_UNATTENTIVE ) ) {
```

File: src/monster.cpp [hunk 4]
```diff
@@ -702,10 +703,14 @@ int monster::print_info( const catacurses::window &w, int vStart, int vLines, in
     mvwprintz( w, point( column, vStart++ ), att.second, att.first );
 
     // Awareness indicator in the third line.
-    bool sees_player = sees( get_player_character() );
-    std::string senses_str = sees_player ? _( "Can see to your current location" ) :
+    Character &pc = get_player_character();
+    bool sees_player = sees( pc );
+    bool player_knows = !pc.has_trait(trait_UNATTENTIVE);
```
Comment (Night-Pryanik) [inline src/monster.cpp:708]:
```suggestion
    const bool player_knows = !pc.has_trait( trait_UNATTENTIVE );
```

Comment (bombasticSlacks) [inline src/monster.cpp:708]:
```suggestion
    bool player_knows = !pc.has_trait( trait_UNATTENTIVE );
```

File: src/monster.cpp [hunk 5]
```diff
@@ -702,10 +703,14 @@ int monster::print_info( const catacurses::window &w, int vStart, int vLines, in
...
+    std::string senses_str = player_knows && sees_player ? _( "Can see to your current location" ) :
                              _( "Can't see to your current location" );
-    vStart += fold_and_print( w, point( column, vStart ), max_width, sees_player ? c_red : c_green,
+    senses_str = !player_knows ? _("You have no idea what is it doing") :
```
Comment (Night-Pryanik) [inline src/monster.cpp:711]:
```suggestion
    senses_str = !player_knows ? _( "You have no idea what is it doing" ) :
```

Comment (bombasticSlacks) [inline src/monster.cpp]:
```suggestion
    senses_str = !player_knows ? _( "You have no idea what is it doing" ) :
```

**Review by bombasticSlacks — COMMENTED** [review]
I'm pretty sure these need cleaned up too

File: src/monster.cpp [hunk 6]
```diff
@@ -702,10 +703,14 @@ int monster::print_info( const catacurses::window &w, int vStart, int vLines, in
     mvwprintz( w, point( column, vStart++ ), att.second, att.first );
 
     // Awareness indicator in the third line.
-    bool sees_player = sees( get_player_character() );
-    std::string senses_str = sees_player ? _( "Can see to your current location" ) :
+    Character &pc = get_player_character();
+    bool sees_player = sees( pc );
+    const bool player_knows = !pc.has_trait( trait_UNATTENTIVE );
+    std::string senses_str = player_knows && sees_player ? _( "Can see to your current location" ) :
```
Comment (GoLoT) [inline src/monster.cpp:709]:
```suggestion
    std::string senses_str = sees_player ? _( "Can see to your current location" ) :
```

Redundant check. If player_knows is false the result of this assignment is overwritten in the next sentence.

**Comment by GoLoT** [comment]
https://github.com/CleverRaven/Cataclysm-DDA/blob/a2b3905db69b688cfe2c9d3fe4578ab6dfc81e1c/src/cata_tiles.cpp#L3464
This needs to check for the trait too, to prevent drawing the visibility (!) marker on monsters.

**Comment by Terrorforge** [comment]
"Unattentive" isn't really a term that anyone uses. The word you are looking for is "Inattentive"

**Comment by mkrutov** [comment]
I've renamed it to *in*attentive, updated tiles bits and implemented Unknown status for shift+v menu(creature list). There is one problem with creature list and I'm unsure what exactly is causing this problem: see screenshot
Screenshot_20211219_230438
.

**Comment by mkrutov** [comment]
Nevermind, I see what is causing it.. sigh.

**Comment by mkrutov** [comment]
LGTM analysis: C/C++ Failing after 72m — Analysis failed (could not build the base commit (579f907))  - unsure what do, this indicates some timeout in one of the tests.

**Comment by GoLoT** [comment]
> LGTM analysis: C/C++ Failing after 72m — Analysis failed (could not build the base commit ([579f907](https://github.com/CleverRaven/Cataclysm-DDA/commit/579f907e424dd640f482b74a0e772fdab774999f))) - unsure what do, this indicates some timeout in one of the tests.

It always times out, you can ignore that one.

**Comment by Zireael07** [comment]
I think you messed something up in the PR - unrelated commits showing up!

**Comment by mkrutov** [comment]
>I think you messed something up in the PR - unrelated commits showing up!

I've rebased it to target latest master, indeed its weird that this commits show up.

**Comment by mqrause** [comment]
Not sure why this is missing the automated labels. You should probably rebase this to have it run the tests again. The labels would give it better visibility to get it merged.

File: src/game.cpp [hunk 7]
```diff
@@ -8108,7 +8118,11 @@ game::vmenu_ret game::list_monsters( const std::vector<Creature *> &monster_list
                         sText = npc_attitude_name( p->get_attitude() );
                         color = p->symbol_color();
                     }
-                    mvwprintz( w_monsters, point( width - 25, y ), color, sText );
+                    if( !player_knows ) {
+                        sText = "Unknown";
```
Comment (Night-Pryanik) [inline src/game.cpp:8122]:
```suggestion
                        sText = _( "Unknown" );
```

**Pull request merged**